nodes.py

```diff
@@ -185,6 +188,9 @@ def descendants(self, selector, **kwargs):
             )
         ]
 
+    def has_parent(self, element, name, attribute, **kwargs):
+        return any(parent.get(attribute[0], "") == attribute[1] for parent in element.parents if parent.name == name)
```

Comment:
Non-blocking: given this doesn't use `self`, perhaps it is better suited as a [static method](https://docs.python.org/3/library/functions.html#staticmethod). I'd also be tempted to make it more specific given it is only used once, e.g.

```suggestion
    @staticmethod
    def has_page_parent(element):
        return any(parent.get("kind", "") == "page" for parent in element.parents if parent.name == "compounddef")
```

And then to call it:

```python
if Node.has_page_parent(element)
```